nomad/structs/structs.go

```diff
@@ -6208,6 +6212,10 @@ func (tg *TaskGroup) Validate(j *Job) error {
 		mErr.Errors = append(mErr.Errors, errors.New("Missing tasks for task group"))
 	}
 
+	if tg.MaxClientDisconnect != nil && *tg.MaxClientDisconnect < 0 {
+		mErr.Errors = append(mErr.Errors, errors.New("max_client_disconnect must be a positive value"))
```

Comment:
Nitpick: 0 isn't positive but is valid. Maybe go with:

```suggestion
		mErr.Errors = append(mErr.Errors, errors.New("max_client_disconnect cannot be negative"))
```

Reply:
Resolved in next push.